### DISCUSSION TITLE
What's the difference between UNet_Learning_Rate?

### DISCUSSION THREAD

**Discussion by meunumerotim333** [TheLastBen/fast-stable-diffusion#1318] [Q&A]
Hello guys,

the earlier version of Dreambooth the main UNet_Learning_Rate was 2e-6. The recommended for that was train 100-200 steps for each picture.

Now the main UNet_Learning_Rate is 1e-5 that now reccomends 65 steps for each picture.

What's the difference between each one? (1e-5 ... 2e-6 ... etc) I haven't find any article about that.

If someone also could explain the importance about Text_Encoder_Training_Steps would be pretty cool to understand how DreamBooth really works.

Thanks :) Hope this discussion helps the community

**Comment by cdefghijkl** [comment]
2e-6 = 0.000002
1e-5 = 0.00001
1e-5 is higher/bigger means it can be faster than 2e-6 to reach the optimal but have to be more careful with steps amount.

TE Training is to teach/introduce your training object token/filename to the model.

That's my knowledge. Cmiiw

**Reply by meunumerotim333** [reply]
Thanks for your answer :)

Do you think the slower method 2e-6 gives better results just to be slower?

**Reply by TheLastBen** [reply]
it depends on the trained subject, for styles in general high learning rate is the way to go

**Reply by meunumerotim333** [reply]
Nice :) I'm training just people faces for friends and family. Like 20 pictures maximum and 8 minimum

**Reply by TheLastBen** [reply]
personally for face, I start with 1e-5, 650 steps unet, then resume the training at 8e-6, 150 steps unet at a time

**Reply by cutecoelho** [reply]
Do you train 650 steps even if is 20 pictures or do you multiply 65 x per photo

Exemple:
20 pictures 1e-5 = 65x20 = 1300

Resume 8e-6 = 15x20 = 300

would be this? or doesn't matter the number of pictures?

**Reply by TheLastBen** [reply]
1e-5 is stronger and higher than 8e-6
for 20 pictures, you can get good results at 1300 steps with 1e-5

**Reply by cutecoelho** [reply]
I've tested here. But is still strong. Isn't blending with the model theme. The results is always realistic pictures.

**Reply by TheLastBen** [reply]
if you set the total steps of the text encoder to 350 at 1e-6, you won't have that issue, make sure you train it only once, after that, train only the unet

**Reply by cutecoelho** [reply]
I've tried exactly like that.
650 1e-5
text encoder 350

resumed 150 8e-6
text encoder 0

I've tested also wihtout resume. Both is too strong, don't blend with model. Only real results. But using 2e-6 is great. I'm using only 2e-6 is saving me.

**Reply by TheLastBen** [reply]
1e-5 is more adequate for V2

**Reply by cutecoelho** [reply]
Ohhh now I undersntand :)

What about 1.5? What's the best configs for face training in your experience? 1.5 still have the best models

**Reply by TheLastBen** [reply]
reduce it at 5e-6 and increase the unet steps to 2000

**Reply by cutecoelho** [reply]
thank you buddy :) I'll try

**Reply by shyt47** [reply]
> reduce it at 5e-6 and increase the unet steps to 2000

Keeping the text encoder to 350 with 2000 steps?

**Reply by TheLastBen** [reply]
yes

**Comment by meunumerotim333** [comment]
Well.. I've tested all the tips, but in my case 2e-6 still the best for train face on model.. because it perfectly fits on the style of the external model.

Does anyone one know what's the unet_learning_rate equivalent of 2e-6 1000 steps?
Because I'm totally dependent of this value.. if sometime the 2e-6 gone I'll be totally lost.. does anyone could give me a tip of an alternative value on a different unet_learning_rate?
